internal/controller/lease_controller.go

```diff
@@ -165,6 +141,194 @@ func (r *LeaseReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 	}
 }
 
+func (r *LeaseReconciler) ReconcileNewLease(
+	ctx context.Context,
+	lease jumpstarterdevv1alpha1.Lease,
+) (ctrl.Result, error) {
+	log := log.FromContext(ctx)
+
+	selector, err := metav1.LabelSelectorAsSelector(&lease.Spec.Selector)
+	if err != nil {
+		log.Error(err, "Error creating selector for label selector")
+		return ctrl.Result{}, err
+	}
+
+	// List all Exporter matching selector
+	var matchedExporters jumpstarterdevv1alpha1.ExporterList
+	if err := r.List(
+		ctx,
+		&matchedExporters,
+		client.InNamespace(lease.Namespace),
+		client.MatchingLabelsSelector{Selector: selector},
+	); err != nil {
+		log.Error(err, "Error listing exporters")
+		return ctrl.Result{}, err
+	}
+
+	onlineExporters := slices.DeleteFunc(
+		matchedExporters.Items,
+		func(exporter jumpstarterdevv1alpha1.Exporter) bool {
+			return !(true &&
+				meta.IsStatusConditionTrue(
+					exporter.Status.Conditions,
+					string(jumpstarterdevv1alpha1.ExporterConditionTypeRegistered),
```

Comment:
? why the true in front
```suggestion
			return !(meta.IsStatusConditionTrue(
					exporter.Status.Conditions,
					string(jumpstarterdevv1alpha1.ExporterConditionTypeRegistered),
```